code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,413 @@
+	if(isliving(user))
+		var/mob/living/person = user
+		id_card = person.get_idcard()
+	if(id_card?.registered_account)
+		current_user = id_card.registered_account
+	if(current_user)
+		data["account_name"] = current_user.account_holder
+
+	return data
+
+/obj/machinery/computer/cryopod/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+
+	var/mob/user = usr
+
+	add_fingerprint(user)
+
+	switch(action)
+		if("one_item")
+			if(!allowed(user))
+				to_chat(user, "<span class='warning'>Access Denied.</span>")
+				return
+			if(!allow_items) return
+
+			if(!params["item"])
+				return
+
+			var/obj/item/item = frozen_items[text2num(params["item"])]
+			if(!item)
+				to_chat(user, "<span class='notice'>\The [item] is no longer in storage.</span>")
+				return
+
+			visible_message("<span class='notice'>The console beeps happily as it disgorges \the [item].</span>")
+			item.forceMove(get_turf(src))
+			frozen_items -= item
+
+		if("all_items")
+			if(!allowed(user))
+				to_chat(user, "<span class='warning'>Access Denied.</span>")
+				return
+			if(!allow_items) return
+
+			visible_message("<span class='notice'>The console beeps happily as it disgorges the desired objects.</span>")
+
+			for(var/obj/item/item in frozen_items)
+				item.forceMove(get_turf(src))
+				frozen_items -= item
+
+	return TRUE
+
+// Cryopods themselves.
+/obj/machinery/cryopod
+	name = "cryogenic freezer"
+	desc = "Suited for Cyborgs and Humanoids, the pod is a safe place for personnel affected by the Space Sleep Disorder to get some rest."
+	icon = 'icons/obj/machines/cryopod.dmi'
+	icon_state = "cryopod-open"
+	density = TRUE
+	anchored = TRUE
+	state_open = TRUE
+
+	var/on_store_message = "has entered long-term storage."
+	var/on_store_name = "Cryogenic Oversight"
+
+	// 3 minutes-ish safe period before being despawned.
+	var/time_till_despawn = 3 MINUTES // This is reduced to 30 seconds if a player manually enters cryo
+	var/despawn_world_time = null          // Used to keep track of the safe period.
+
+	var/obj/machinery/computer/cryopod/control_computer
+	COOLDOWN_DECLARE(last_no_computer_message)
+
+	// These items are preserved when the process() despawn proc occurs.
+	var/static/list/preserve_items = list(
+		/obj/item/hand_tele,
+		/obj/item/card/id/advanced/gold/captains_spare,
+		/obj/item/aicard,
+		/obj/item/mmi,
+		/obj/item/paicard,
+		/obj/item/gun,
+		/obj/item/pinpointer,
+		/obj/item/clothing/shoes/magboots,
+		/obj/item/areaeditor/blueprints,
+		/obj/item/clothing/head/helmet/space,
+		/obj/item/clothing/suit/space,
+		/obj/item/clothing/suit/armor,
+		/obj/item/defibrillator/compact,
+		/obj/item/reagent_containers/hypospray/cmo,
+		/obj/item/clothing/accessory/medal/gold/captain,
+		/obj/item/clothing/gloves/krav_maga,
+		/obj/item/nullrod,
+		/obj/item/tank/jetpack,
+		/obj/item/documents,
+		/obj/item/nuke_core_container
+	)
+	// These items will NOT be preserved
+	var/static/list/do_not_preserve_items = list (
```

Comment:
If these are the two variables, "preserve and not preserve", what's in the middle? Of items that aren't in either list?

Reply:
Items that are outright deleted once the person steps in